### PR TITLE
feat(modeling): new scission() function added to booleans

### PR DESCRIPTION
**Pull request by z3dev**
These changes add a new function to the booleans, called scission(). This new function cuts/divideds/scissions the given geometry into component pieces.

This is useful to pull apart larger designs into pieces, especially STL mesh imports. As well as special times when the results of boolean operations need to be split into pieces. See #517 splitting of solid into pieces 

BIG THANKS to @giboneet for the initial implementation.

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

### CODE REVIEW

File: packages/modeling/src/operations/booleans/scission.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const flatten = require('../../utils/flatten')
+
+// const geom2 = require('../../geometries/geom2')
```
Comment (mister-jscad) [inline packages/modeling/src/operations/booleans/scission.js:3]: Hmm... is scission of 2D geometries possible?

File: packages/modeling/src/operations/booleans/scissionGeom3.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  const polygonIndexesPerPoint = new Map()
+  const temp = vec3.create()
+  polygons.forEach((polygon, polygonIndex) => {
+    polygon.vertices.forEach((point) => {
+      insertMapping(polygonIndexesPerPoint, vec3.snap(temp, point, eps), polygonIndex)
+    })
+  })
+
+  const zz = polygons.map((polygon, polygonIndex) => {
```
Comment (mister-jscad) [inline packages/modeling/src/operations/booleans/scissionGeom3.js:39]: zz doesn't mean much. how about renaming to something like 'indexesPerPolygon'?

File: packages/modeling/src/operations/booleans/scissionGeom3.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    const indexes = []
+    polygon.vertices.forEach((point) => {
+      indexes.push(...findMapping(polygonIndexesPerPoint, vec3.snap(temp, point, eps)))
```
Comment (mister-jscad) [inline packages/modeling/src/operations/booleans/scissionGeom3.js:42]: Maybe concat() is a better choice as there are limits on the size of arrays using '...'

File: packages/modeling/src/operations/booleans/scissionGeom3.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        })
+      } while (merges > 0)
+      mapi.indexes = indexes
+    }
+  }
+
+  // construit le tableau des geometry à retourner
+  // build array of geometry to return
+  const newgeometries = []
+  const zzl = zz.length
```
Comment (mister-jscad) [inline packages/modeling/src/operations/booleans/scissionGeom3.js:81]: zzl could be reused above

**Review by mister-jscad — CHANGES_REQUESTED** [review]
Suggested a few changes.

File: packages/modeling/src/operations/booleans/scission.js (hunk 1 above)
Comment (z3dev) [reply packages/modeling/src/operations/booleans/scission.js:3]: It's probably possible, but not straight forward.

Should it be implemented now? Not sure.

**Review by mister-jscad — APPROVED** [review]
The changes look fine.

**Pull request merged**